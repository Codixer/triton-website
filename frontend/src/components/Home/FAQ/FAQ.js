import React, { Component } from "react";
import styles from "./FAQ.scss";
import OpenSansTypography from "../../OpenSansTypography/OpenSansTypography";
import Container from "../../Container/Container";
import Question from "./Question/Question";
import { Grid } from "@rmwc/grid";
import "@material/layout-grid/dist/mdc.layout-grid.css";

class FAQ extends Component {
  render() {
    return (
      <Container id="faq">
        <OpenSansTypography use="headline4">Frequently Asked Questions</OpenSansTypography>
        <Grid className={styles.questionContainer}>
          <Question title="If the API is free, why would I pay for the plugin?">
            The API does not contain any plugin code or functionality, it simply contains the
            documentation and names of methods - allowing a developer to compile their own plugin.
            The API does nothing by itself and will not function as a plugin.
          </Question>
          <Question title="Where can I request features?">
            You can open an "issue" on our{" "}
            <a href="/github_issues" target="_blank">
              GitHub
            </a>{" "}
            or visit our{" "}
            <a href="/discord" target="_blank">
              Discord server
            </a>
            .
          </Question>
          <Question title="Where are the commands/permissions and installation instructions?">
            They are in our documentation. Take a look <a href="/docs">here</a>.
          </Question>
          <Question title="When I try to use specials characters like (ãáàçñ, etc) on the languages.json file, some weird characters appear.">
            There is a problem with the encoding of the file. Try another encoding.
          </Question>
          <Question title="The plugin isn't working for me or I have some issue.">
            For any questions, join our{" "}
            <a href="/discord" target="_blank">
              Discord
            </a>{" "}
            server. If you've found a bug, please open an issue on our{" "}
            <a href="/github_issues" target="_blank">
              GitHub
            </a>
            .
          </Question>
        </Grid>
      </Container>
    );
  }
}

export default FAQ;
