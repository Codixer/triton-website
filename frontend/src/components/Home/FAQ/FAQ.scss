:local(.questionContainer) {
  margin-top: 15px;
}
