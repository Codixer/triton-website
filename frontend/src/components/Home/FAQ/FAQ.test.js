import React from "react";
import FAQ from "./FAQ";
import renderer from "react-test-renderer";

it("<FAQ /> renders correctly", () => {
  const tree = renderer.create(<FAQ />).toJSON();
  expect(tree).toMatchSnapshot();
});
