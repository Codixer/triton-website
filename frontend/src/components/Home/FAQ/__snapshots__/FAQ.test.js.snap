// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FAQ /> renders correctly 1`] = `
<div
  id="faq"
>
  <span
    className="mdc-typography--headline4"
  >
    Frequently Asked Questions
  </span>
  <div
    className="mdc-layout-grid"
  >
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell mdc-layout-grid__cell--span-4"
      >
        <div
          className="mdc-elevation--z8"
        >
          <div>
            <i
              className="rmwc-icon material-icons"
            >
              comment-question.svg
            </i>
            <span
              className="mdc-typography--headline6"
            >
              If the API is free, why would I pay for the plugin?
            </span>
          </div>
          <br />
          <span
            className="mdc-typography--body1"
          >
            The API does not contain any plugin code or functionality, it simply contains the documentation and names of methods - allowing a developer to compile their own plugin. The API does nothing by itself and will not function as a plugin.
          </span>
        </div>
      </div>
      <div
        className="mdc-layout-grid__cell mdc-layout-grid__cell--span-4"
      >
        <div
          className="mdc-elevation--z8"
        >
          <div>
            <i
              className="rmwc-icon material-icons"
            >
              comment-question.svg
            </i>
            <span
              className="mdc-typography--headline6"
            >
              Where can I request features?
            </span>
          </div>
          <br />
          <span
            className="mdc-typography--body1"
          >
            You can open an "issue" on our
             
            <a
              href="/github_issues"
              target="_blank"
            >
              GitHub
            </a>
             
            or visit our
             
            <a
              href="/discord"
              target="_blank"
            >
              Discord server
            </a>
            .
          </span>
        </div>
      </div>
      <div
        className="mdc-layout-grid__cell mdc-layout-grid__cell--span-4"
      >
        <div
          className="mdc-elevation--z8"
        >
          <div>
            <i
              className="rmwc-icon material-icons"
            >
              comment-question.svg
            </i>
            <span
              className="mdc-typography--headline6"
            >
              Where are the commands/permissions and installation instructions?
            </span>
          </div>
          <br />
          <span
            className="mdc-typography--body1"
          >
            They are in our documentation. Take a look 
            <a
              href="/docs"
            >
              here
            </a>
            .
          </span>
        </div>
      </div>
      <div
        className="mdc-layout-grid__cell mdc-layout-grid__cell--span-4"
      >
        <div
          className="mdc-elevation--z8"
        >
          <div>
            <i
              className="rmwc-icon material-icons"
            >
              comment-question.svg
            </i>
            <span
              className="mdc-typography--headline6"
            >
              When I try to use specials characters like (ãáàçñ, etc) on the languages.json file, some weird characters appear.
            </span>
          </div>
          <br />
          <span
            className="mdc-typography--body1"
          >
            There is a problem with the encoding of the file. Try another encoding.
          </span>
        </div>
      </div>
      <div
        className="mdc-layout-grid__cell mdc-layout-grid__cell--span-4"
      >
        <div
          className="mdc-elevation--z8"
        >
          <div>
            <i
              className="rmwc-icon material-icons"
            >
              comment-question.svg
            </i>
            <span
              className="mdc-typography--headline6"
            >
              The plugin isn't working for me or I have some issue.
            </span>
          </div>
          <br />
          <span
            className="mdc-typography--body1"
          >
            For any questions, join our
             
            <a
              href="/discord"
              target="_blank"
            >
              Discord
            </a>
             
            server. If you've found a bug, please open an issue on our
             
            <a
              href="/github_issues"
              target="_blank"
            >
              GitHub
            </a>
            .
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
